### ISSUE TITLE
Got NAN when calling embedding_lookup_sparse with weights and "mean" combiner

### ISSUE THREAD

**Issue by zhangyafeikimi** [tensorflow/tensorflow#14851]
### System information
- **OS Platform and Distribution**: Win7 64bit
- **TensorFlow installed from**: anaconda binary
- **TensorFlow version**: 1.2.1
- **Python version**: 3.6

### Describe the problem
I always got nan when I call embedding_lookup_sparse with weights and "mean" combiner.
Code pieces are listed below:

### Source code / logs
from __future__ import absolute_import
from __future__ import print_function
import os
import numpy as np
import tensorflow as tf
os.environ['TF_CPP_MIN_LOG_LEVEL'] = '3'

a = np.arange(192).reshape(24, 8)
print(a)

a = tf.Variable(a, dtype=tf.float32)
ids = tf.SparseTensor(
    indices=[[1, 0], [2, 0], [2, 1], [2, 2], [3, 0], [9, 1]],
    values=[10, 1, 2, 3, 4, 5],
    dense_shape=[1, 1])
weights = tf.SparseTensor(
    indices=[[1, 0], [2, 0], [2, 1], [2, 2], [3, 0], [9, 1]],
    values=[1, 0.2, 0.4, 0.4, 1, 1],
    dense_shape=[1, 1])
b = tf.nn.embedding_lookup_sparse(
    a, ids, weights, partition_strategy='mod', combiner='sum')
c = tf.pad(b, [[0, 16 - tf.shape(b)[0]], [0, 0]], mode='CONSTANT')

sess = tf.Session()
sess.run(tf.global_variables_initializer())
[value] = sess.run([c])
print(value)

**Comment by printdhruv** [comment]
`[
[  0.           0.           0.           0.           0.           0.
    0.           0.        ]
 [ 80.          81.          82.          83.          84.          85.
   86.          87.        ]
 [ 17.60000038  18.60000038  19.60000229  20.60000038  21.59999847
   22.60000229  23.60000038  24.60000038]
 [ 32.          33.          34.          35.          36.          37.
   38.          39.        ]
 [  0.           0.           0.           0.           0.           0.
    0.           0.        ]
 [  0.           0.           0.           0.           0.           0.
    0.           0.        ]
 [  0.           0.           0.           0.           0.           0.
    0.           0.        ]
 [  0.           0.           0.           0.           0.           0.
    0.           0.        ]
 [  0.           0.           0.           0.           0.           0.
    0.           0.        ]
 [ 40.          41.          42.          43.          44.          45.
   46.          47.        ]
 [  0.           0.           0.           0.           0.           0.
    0.           0.        ]
 [  0.           0.           0.           0.           0.           0.
    0.           0.        ]
 [  0.           0.           0.           0.           0.           0.
    0.           0.        ]
 [  0.           0.           0.           0.           0.           0.
    0.           0.        ]
 [  0.           0.           0.           0.           0.           0.
    0.           0.        ]
 [  0.           0.           0.           0.           0.           0.
    0.           0.        ]
]`

Is above out put correct?

**Comment by zhangyafeikimi** [comment]
sorry, use combiner="mean", and retry.

**Comment by facaiy** [comment]
I think `np.nan` is reasonable, because some lines, say 0, 4, 5, 6, 7, 8, are missing in
` indices=[[1, 0], [2, 0], [2, 1], [2, 2], [3, 0], [9, 1]],`. For those lines, value is 0, and sum is 0, so `0 / 0 = np.nan`.

**Comment by facaiy** [comment]
```python
import numpy as np
import tensorflow as tf

a = np.arange(192).reshape(24, 8)
print(a)

a = tf.Variable(a, dtype=tf.float32)
ids = tf.SparseTensor(
indices=[[1, 0], [2, 0], [2, 1], [2, 2], [3, 0], [9, 1]],
values=[10, 1, 2, 3, 4, 5],
dense_shape=[1, 1])
weights = tf.SparseTensor(
indices=[[1, 0], [2, 0], [2, 1], [2, 2], [3, 0], [9, 1]],
values=[1, 0.2, 0.4, 0.4, 1, 1],
dense_shape=[1, 1])
b = tf.nn.embedding_lookup_sparse(
a, ids, weights, partition_strategy='mod', combiner='mean')

sess = tf.Session()
sess.run(tf.global_variables_initializer())
[value] = sess.run([b])

np.set_printoptions(precision=2)
print(value)
print(value.shape)
```

```bash
[[  nan   nan   nan   nan   nan   nan   nan   nan]
 [ 80.   81.   82.   83.   84.   85.   86.   87. ]
 [ 17.6  18.6  19.6  20.6  21.6  22.6  23.6  24.6]
 [ 32.   33.   34.   35.   36.   37.   38.   39. ]
 [  nan   nan   nan   nan   nan   nan   nan   nan]
 [  nan   nan   nan   nan   nan   nan   nan   nan]
 [  nan   nan   nan   nan   nan   nan   nan   nan]
 [  nan   nan   nan   nan   nan   nan   nan   nan]
 [  nan   nan   nan   nan   nan   nan   nan   nan]
 [ 40.   41.   42.   43.   44.   45.   46.   47. ]]
(10, 8)
```

**Comment by zhangyafeikimi** [comment]
facaiy, exactly right.

as its name "sparse", missing values are common cases.

how can nan be spread everywhere?

**Comment by facaiy** [comment]
`tf.where` can help you clean all `np.nan`:
```python
a, ids, weights, partition_strategy='mod', combiner='mean')
# clean np.nan
b = tf.where(tf.is_nan(b), tf.zeros_like(b), b)
c = tf.pad(b, [[0, 16 - tf.shape(b)[0]], [0, 0]], mode='CONSTANT')
```

**Comment by zhangyafeikimi** [comment]
@facaiy    thanks.

tf.where is not an elegant way, but better than nothing.

**Comment by tatatodd** [comment]
@facaiy Please update this bug when your PR #14865 lands.  And thanks for your help and PRs!  :)

**Comment by tensorflowbutler** [comment]
It has been 14 days with no activity and the `awaiting response` label was assigned. Is this still an issue? Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
It has been 14 days with no activity and the `awaiting response` label was assigned. Is this still an issue? Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Awaiting Response: It has been 14 days with no activityand the `awaiting response` label was assigned. Is this still an issue?

**Comment by tensorflowbutler** [comment]
Nagging Awaiting Response: It has been 14 days with no activityand the `awaiting response` label was assigned. Is this still an issue?

**Comment by sachinprasadhs** [comment]
You seem to be using older version(1.x) of Tensorflow which is not supported anymore. Please try the latest [Tensorflow version](https://www.tensorflow.org/install/pip) and let us know if the problem still persists.
Check the `[tf.nn.embedding_lookup_sparse](https://www.tensorflow.org/api_docs/python/tf/nn/embedding_lookup_sparse)` in Tensorflow 2 version for the changes. Thanks!
